pkg/messaging/nats/publisher.go

```diff
@@ -22,7 +22,7 @@ type publisher struct {
 
 // NewPublisher returns NATS message Publisher.
 func NewPublisher(url string) (messaging.Publisher, error) {
-	conn, err := broker.Connect(url)
+	conn, err := broker.Connect(url, broker.MaxReconnects(-1))
```

Comment:
I would define this constant somewhere and give it a self-documenting name, like `tryForever`, or you can leave a comment what this `-1` means (for someone reading the code for the first time)